### DISCUSSION TITLE
Solidity-Metrics extension doesn't work on Windows

### DISCUSSION THREAD

**Discussion by pymachado** [Cyfrin/security-and-auditing-full-course-s23#47] [Q&A] [answered]
Hi everyone.  I have problems with solidity-metrics extension, it doesn't work properly in my VS Code enviroment. When I tried to run it a notification error appear and the description is "No valid solidity source files found.". Maybe someone had the same bug and can help me to solve it. Thank you. 
![image](https://github.com/Cyfrin/security-and-auditing-full-course-s23/assets/63087258/554f19ae-638b-48a1-aef7-271100bbc02c)

**Comment by rasyidred** [comment, marked as answer]
This is a workaround for me. Open the command palette (`ctrl + shift + P`), then type `Solidity Metric: report metrics for all open workspaces`. However, this action will give you the report for almost all of the `.sol` files in your local repo, e.g., inside `src` and `script` folders if you use Foundry.

**Reply by cromewar** [reply]
@pymachado was this helpful to you?

**Reply by pymachado** [reply]
This works to me. Thank you @rasyidred

**Reply by pymachado** [reply]
Yes it was.

**Reply by pymachado** [reply]
![image](https://github.com/Cyfrin/security-and-auditing-full-course-s23/assets/63087258/15f95d2e-d228-45fe-a560-51726af17d4c)

**Reply by rasyidred** [reply]
Glad to help!

**Reply by Afriauditor** [reply]
this was helpful mate thanks

**Reply by clement-touzet** [reply]
thank you @rasyidred

**Reply by Nightfury456** [reply]
you can do ```ctrl + shift + p``` and search ```solidity metrices: Report metrices for current open file``` for the single file

**Reply by EngrPips** [reply]
Interesting.

**Comment by park1731** [comment]
make a separate folder of scope files and open command pallet and run solidity metrics for entire workspace. It will give report for only file which are in scope as you made a separate file

**Reply by EngrPips** [reply]
Interesting.
